ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -33,6 +33,33 @@ public enum VibrationSupport {
   case taptic // Has the Taptic engine, allowing use of UIImpactFeedbackGenerator for customizable vibrations
 }
 
+public enum SpacebarText {
+  case LANGUAGE
+  case KEYBOARD
+  case LANGUAGE_KEYBOARD
+  case BLANK
+
+  // Maps to enum SpacebarText in kmwbase.ts
+  public static func fromString(mode: String) -> SpacebarText {
+    switch(mode) {
+      case "language": return LANGUAGE
+      case "keyboard": return KEYBOARD
+      case "languageKeyboard": return LANGUAGE_KEYBOARD
+      case "blank": return BLANK
+      default: return LANGUAGE_KEYBOARD
+    }
+  }
+
+  public func toString() -> String {
+    switch(self) {
+    case .LANGUAGE: return "language"
+    case .KEYBOARD: return "keyboard"
+    case .LANGUAGE_KEYBOARD: return "languageKeyboard"
+    case .BLANK: return "blank"
+    }
+  }
+};
```

Comment:
1.  Why in `Manager` instead of in `InputViewController`?
2. You might find this to be a lot more convenient...

https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html#ID535

One example use within the codebase:

https://github.com/keymanapp/keyman/blob/1c1a9bcbb818e342f3c86940746124fad2f4d4c0/ios/engine/KMEI/KeymanEngine/Classes/KeymanPackage.swift#L12-L27

Also note the "Initializing from a Raw Value" section after the linked one; it can work both ways.

One of the things I like about Swift is just how much leeway it gives enums.  _So_ much nicer than many other languages I've coded with.

Reply:
In `Manager` where there are other enums as well. Yes, I was aware that the enum format was not as tight as it could be. Happy to adjust it.